File: Day 27/Solution 2.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int count = 0;
        int i = 0;
        while(i<n-2&&nums[i]==0)i++;
        for(int k=n-1; k>1; k--){
            int r = k-1;
            int l = i;
            while(l<r){
                while(l<r&&nums[l]+nums[r]<=nums[k])l++;
                count += r-l;
                r--;
        }
        }
        return count;
}
};

File: Day 100/Solution 2.py
class Solution:
    def thirdMax(self, nums):
        """
            :type nums: List[int]
            :rtype: int
            """
        heap = [-n for n in set(nums)]
        heapq.heapify(heap)
        res = []
        count = 0
        print(heap)
        while heap and count < 3:
            res.append(-heapq.heappop(heap))
            count += 1
        return res[2] if len(res) == 3 else res[0]
